File: 剑指Offer/斐波那契数列/arrayList.java
public class Solution {
    public int Fibonacci(int n) {
        if(n<2){
            return n;
        }
        int[] res = new int[n+1];
        res[0] = 0;
        res[1] = 1;
        for(int i=2;i<n+1;i++){
            res[i] = res[i-1]+res[i-2];
        }
        return res[n];
    }
}

File: 剑指Offer/斐波那契数列/arrayList.py
# -*- coding:utf-8 -*-
class Solution:
    def Fibonacci(self,n):
    # write code here
        if n<2:return n
        res = [0 for i in range(0,n+1)]
        res[0],res[1] = 0,1
        for i in range(2,n+1):
            res[i] = res[i-1] + res[i-2]
        return res[-1]